workflows/builtin: published selection-math.ts fails consumer tsc under noUncheckedIndexedAccess

### What happened?

`@bastani/atomic@0.9.14-alpha.5` ships `packages/workflows/builtin/selection-math.ts` as the `import` target for `@bastani/atomic/workflows/builtin` (and therefore for the ambient `@bastani/workflows` bridge). A consumer that type-checks those published `.ts` files with `noUncheckedIndexedAccess` fails:

```
node_modules/@bastani/atomic/dist/builtin/workflows/builtin/selection-math.ts(130,3): error TS2532: Object is possibly 'undefined'.
node_modules/@bastani/atomic/dist/builtin/workflows/builtin/selection-math.ts(131,3): error TS2532: Object is possibly 'undefined'.
node_modules/@bastani/atomic/dist/builtin/workflows/builtin/selection-math.ts(132,3): error TS2532: Object is possibly 'undefined'.
node_modules/@bastani/atomic/dist/builtin/workflows/builtin/selection-math.ts(133,3): error TS2532: Object is possibly 'undefined'.
```

The four writes are in `accumulate`:

```ts
export function accumulate(prefs: readonly Preference[], w: number[], c: number[]): void {
	for (const { a, b, p } of prefs) {
		w[a] += p;
		c[a] += 1;
		w[b] += 1 - p;
		c[b] += 1;
	}
}
```

Under `noUncheckedIndexedAccess`, `w[a]` is `number | undefined`, so `+=` is illegal. The same file already uses non-null assertions for other indexed reads (`pivots[first]!`, `w[index]! / c[index]!`); this function just missed them.

This is not caught by the repo's own `tsc --noEmit`: root `tsconfig.json` is `strict` but does **not** set `noUncheckedIndexedAccess`. Existing `test/unit/selection-math.test.ts` covers the runtime contract and stays green.

`0.9.14-alpha.4` does not ship this file, so this is a regression on the 0.9.14-alpha.5 tournament/PPT landing (related: #2488).

`skipLibCheck` does not help. The published export is:

```json
"./workflows/builtin": {
  "types": "./dist/builtin/workflows/builtin/index.d.ts",
  "import": "./dist/builtin/workflows/builtin/index.ts"
}
```

A consumer with `allowImportingTsExtensions` (or one that otherwise follows `.js` re-exports to sibling `.ts`) type-checks the implementation, not just the `.d.ts`. Importing `@bastani/atomic` is enough: the root types pull `./workflows/ambient`, which enumerates every builtin, which pulls `tournament` → `selection-math.ts`.

### Steps to reproduce

Minimal throwaway consumer against the published package (no Atomic checkout required):

```bash
tmpdir=$(mktemp -d)
cd "$tmpdir"

cat > package.json <<'JSON'
{
  "name": "atomic-nui-repro",
  "private": true,
  "type": "module",
  "devDependencies": {
    "@bastani/atomic": "0.9.14-alpha.5",
    "typescript": "^5.7.3"
  }
}
JSON

cat > tsconfig.json <<'JSON'
{
  "compilerOptions": {
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "target": "ES2022",
    "strict": true,
    "noUncheckedIndexedAccess": true,
    "allowImportingTsExtensions": true,
    "skipLibCheck": true,
    "noEmit": true
  },
  "include": ["src/**/*.ts"]
}
JSON

mkdir src
cat > src/index.ts <<'TS'
import { VERSION } from "@bastani/atomic";
export const version: string = VERSION;
TS

npm install
npx tsc --noEmit
```

Expected today: four `TS2532` errors in `selection-math.ts` lines 130–133. `atomic -ne` is not involved; this is a published-types failure, not an extension-runtime failure.

### Expected behavior

An installed `@bastani/atomic` must type-check for a consumer using `strict` + `noUncheckedIndexedAccess` + `allowImportingTsExtensions` + `skipLibCheck`, with no `paths` alias and no `declare module` shim. That is the #1208 contract.

Smallest source fix — match the rest of the file:

```ts
export function accumulate(prefs: readonly Preference[], w: number[], c: number[]): void {
	for (const { a, b, p } of prefs) {
		w[a]! += p;
		c[a]! += 1;
		w[b]! += 1 - p;
		c[b]! += 1;
	}
}
```

(`a`/`b` are candidate indices the caller already sized `w`/`c` for; the existing unit test in `test/unit/selection-math.test.ts` already asserts that contract.)

### How to lock the fix

The right regression test is **not** another runtime case in `selection-math.test.ts`. It is a new **pass** variant in `packages/coding-agent/scripts/verify-workflow-sdk-types.ts`, which already packs `@bastani/atomic` and type-checks throwaway consumers outside the repo.

Add a variant whose `tsconfig()` extra is:

```ts
{
  noUncheckedIndexedAccess: true,
  allowImportingTsExtensions: true,
}
```

Reuse the existing `auto-include-via-atomic-import` files (import `@bastani/atomic` so ambient pulls every builtin, including `tournament` / `selection-math`). Expect `tsc --noEmit` to pass.

Verify:

```bash
# 1. Confirm the bug on current tree / published tarball
cd packages/coding-agent
SKIP_BUILD=1 bun run verify:workflow-types   # existing variants still pass
# then run the new variant; it should FAIL before the source fix

# 2. After the accumulate patch
bun run verify:workflow-types                # rebuilds, then all variants pass
npm run test:unit -- test/unit/selection-math.test.ts
npm run check
```

If you want a checkout-only smoke test before touching the fixture script:

```bash
npx tsc --noEmit --pretty false \
  --strict --noUncheckedIndexedAccess --allowImportingTsExtensions \
  --module NodeNext --moduleResolution NodeNext --skipLibCheck --noEmit \
  packages/workflows/builtin/selection-math.ts
```

That should report the same four `TS2532`s before the patch and exit 0 after.

Optional follow-up, not required to close this: stop publishing `.ts` as the `import` target for `./workflows/builtin` so consumers stay on `.d.ts` + `skipLibCheck`. The four-line `accumulate` patch is enough for 0.9.14-alpha.6.

### Version

0.9.14-alpha.5

--- a/packages/coding-agent/scripts/verify-workflow-sdk-types.ts
+++ b/packages/coding-agent/scripts/verify-workflow-sdk-types.ts
@@ -133,6 +133,17 @@ const VARIANTS: readonly Variant[] = [
 		},
 	},
 	{
+		name: "auto-include-no-unchecked-indexed-access",
+		description: "Regression lock for issue #2543: auto-included workflow types remain type-safe with noUncheckedIndexedAccess.",
+		expect: "pass",
+		files: {
+			"package.json": consumerPackageJson("ext-and-workflow-no-unchecked-indexed-access", { extensions: ["./src/index.ts"], workflows: ["./workflows"] }),
+			"tsconfig.json": tsconfig({ noUncheckedIndexedAccess: true, allowImportingTsExtensions: true }),
+			"src/index.ts": `import { VERSION } from "@bastani/atomic";\nexport const myExtensionVersion: string = VERSION;\n`,
+			"workflows/hello.ts": WORKFLOW_FILE,
+		},
+	},
+	{
 		name: "negative-no-optin",
 		description: "Negative control: pure workflow-only package with NO opt-in must fail with TS2307 (the bug this fix closes).",
 		expect: "fail",

--- a/packages/workflows/CHANGELOG.md
+++ b/packages/workflows/CHANGELOG.md
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 
 ## [Unreleased]
 
+### Fixed
+
+- Fixed consumer TypeScript errors in `selection-math` when `noUncheckedIndexedAccess` is enabled ([#2543](https://github.com/bastani-inc/atomic/issues/2543)).
+
 ## [0.9.14-alpha.5] - 2026-08-19
 
 ### Added

--- a/packages/workflows/builtin/selection-math.ts
+++ b/packages/workflows/builtin/selection-math.ts
@@ -127,10 +127,10 @@ export function soft_win(scoreA: number, scoreB: number): number {
 /** Add each preference to both candidates' win mass and comparison count. */
 export function accumulate(prefs: readonly Preference[], w: number[], c: number[]): void {
 	for (const { a, b, p } of prefs) {
-		w[a] += p;
-		c[a] += 1;
-		w[b] += 1 - p;
-		c[b] += 1;
+		w[a]! += p;
+		c[a]! += 1;
+		w[b]! += 1 - p;
+		c[b]! += 1;
 	}
 }
 

